src/main/jbake/content/blog/2016/10/14/hawkular-apm-polyglot-zipkin.adoc

```diff
@@ -0,0 +1,200 @@
+
+Execute following requests which we will analyze in the following sections.
+-----
+1. curl -ivX GET 'http://localhost:3001/nodejs/hello'
+2. curl -ivX GET 'http://localhost:3000/dropwizard/A
+3. curl -ivX GET 'http://localhost:3000/dropwizard/asyncTwoOutgoingCalls'
+4. curl -ivX POST -H 'Content-Type: application/json' 'http://localhost:3001/nodejs/createUser' -d '{"name": "jdoe"}'
+-----
+
+== Hawkular APM
+In Hawkular APM user interface we are going to look at distributed tracing tab. If you run previous 
+commands there should be some data for analysis. 
+
+Following figure shows aggregated service interactions for selected initial endpoint (fourth request). 
+Nodes represent endpoint with URL and edges invocation flow.
+ 
+Nodes holds statistics information like average, minimum and maximum duration, 
+total number of executions, service name and invoked URL. 
+Edges contain timing information of a network latency. This screen shows only 
+nodes with a URL. Databases and other components are shown in trace detail view.
+
+For query capabilities we can specify time span, business transaction 
+(currently available only for java agent instrumentation and OpenTracing provider) and properties. 
+Properties represent binary annotations.
+
+ifndef::env-github[]
+image::/img/blog/2016/apm-distributed-tracing-createUser.png[title="Hawkular APM Distributed Tracing",align="center"]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2016/apm-distributed-tracing-createUser.png[title="Hawkular APM Distributed 
+Tracing",
+align="center"]
+endif::[]
+
+Next screen shows instance details. In other words a list of traces for the selected initial endpoint.
+Table contains basic information as collected properties, start time, principal(for java agent instrumentation) 
+and overall trace duration. Details button opens a detailed graph.
+
+ifndef::env-github[]
+image::/img/blog/2016/apm-instance-details.png[title="Hawkular APM Instance Details",align="center"]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2016/apm-instance-details.png[title="Hawkular APM Instance Details",align="center"]
+endif::[]
+
+In the detailed view users can see different types of nodes: producers (client span), 
+consumers (server span) and components (databases, EJBs). Tooltip is showing recorded binary annotations. 
+In this case it is a CQL query for storing an user into Apache Cassandra.
+
+This concrete graph starts with a consumer (REST handler) and produces two producers (outgoing requests). These
+requests are consumed by dropwizard and wildfly-swarm application and so on.
+
+ifndef::env-github[]
+image::/img/blog/2016/apm-instance-details-createUser.png[title="Hawkular APM Instance Details",align="center"]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2016/apm-instance-details-createUser.png[title="Hawkular APM Instance Details",align="center"]
+endif::[]
+
+== Zipkin
```

Comment:
The idea itself is good.
But when you read the text and get to the subtitle of "Zipkin" followed by "Please start Zipkin server and..."
the reader does not get the idea that it is a comparison, but rather that he needs both to get stuff done (what she could also use with Zipkin alone)